### DISCUSSION TITLE
Weird Turbo Schedule type

### DISCUSSION THREAD

**Discussion by pinea00** [Panchovix/stable-diffusion-webui-reForge#52] [Q&A] [answered]
The original FORGE DPM++ 2M SDE turbo only needs 8 STEP to produce pictures with the same quality as DPM++ 2M KArras 20 step.
However, reforge uses DPM++ 2M SDE SD turbo 8step to generate images that appear to have not yet been denoised.
The strange thing is that reforge DPM++ 2M SDE Karras can produce the same quality pictures as Forge DPM++ 2M SDE turbo 8 step in 8 steps.
Usage model: dreamsharpXL turbo Code:refoge main
![00022-1942144205](https://github.com/user-attachments/assets/1a412bac-e6b4-4965-acfa-d065f59f314a)

**Comment by Panchovix** [comment]
I have noticed this as well, gonna re-add the Stock forge ones but naming it to "Euler A Turbo Forge", and similar like that.

**Comment by Panchovix** [comment]
Done, restored the 3 Turbo samplers, and removed SD Turbo scheduler.

**Reply by pinea00** [reply]
very good, thank you
May I ask more something? ~Is reforge’s support for SD3 incomplete? Loading sd3_medium.safetensors will cause an error and crash (the clip has been added and downloaded first)

**Reply by Panchovix** [reply]
Is not complete, since well all the code to make SD3 work is on reForge (dev_upstream), but I have to modify forge_loader and unet_patcher (both on modules_forge folder) to make SD3 work, which I haven't been tinkering or managed to do yet.

**Reply by pinea00** [reply]
Just tested Turbo samplers and got an error and couldn't work
------------------------------------------------------------
Traceback (most recent call last):
  File "S:\stable-diffusion-webui-reForge\modules_forge\main_thread.py", line 37, in loop
    task.work()
  File "S:\stable-diffusion-webui-reForge\modules_forge\main_thread.py", line 26, in work
    self.result = self.func(*self.args, **self.kwargs)
  File "S:\stable-diffusion-webui-reForge\modules\txt2img.py", line 110, in txt2img_function
    processed = processing.process_images(p)
  File "S:\stable-diffusion-webui-reForge\modules\processing.py", line 815, in process_images
    res = process_images_inner(p)
  File "S:\stable-diffusion-webui-reForge\modules\processing.py", line 988, in process_images_inner
    samples_ddim = p.sample(conditioning=p.c, unconditional_conditioning=p.uc, seeds=p.seeds, subseeds=p.subseeds, subseed_strength=p.subseed_strength, prompts=p.prompts)
  File "S:\stable-diffusion-webui-reForge\modules\processing.py", line 1362, in sample
    samples = self.sampler.sample(self, x, conditioning, unconditional_conditioning, image_conditioning=self.txt2img_image_conditioning(x))
  File "S:\stable-diffusion-webui-reForge\modules_forge\forge_alter_samplers.py", line 105, in sample
    return super().sample(p, x, conditioning, unconditional_conditioning, steps, image_conditioning)
  File "S:\stable-diffusion-webui-reForge\modules\sd_samplers_kdiffusion.py", line 224, in sample
    sigmas = self.get_sigmas(p, steps).to(x.device)
AttributeError: 'NoneType' object has no attribute 'to'
'NoneType' object has no attribute 'to'
*** Error completing request
*** Arguments: ('task(o0shg8id6sv4q1u)', <gradio.routes.Request object at 0x000001686A1C9D50>, '1 girl, brown medium-length hair, braids, bangs, Taiwanese, wearing a high school girl uniform, white clothes, pink pleated group, sitting on a bench in the park, morning sunshine <lora:add-detail-xl:1> Obvious and sharp details', 'abnormal limbs', [], 1, 1, 2, 1024, 1024, False, 0.7, 2, 'Latent', 0, 0, 0, 'Use same checkpoint', 'Use same sampler', 'Use same scheduler', '', '', [], 0, 8, 'DPM++ 2M SDE Turbo', 'Automatic', False, '', 0.8, -1, False, -1, 0, 0, 0, True, False, False, False, False, 'base', 'None', 3, 5, 2, 0, 0.1, 3, 5, False, 0, <scripts.animatediff_ui.AnimateDiffProcess object at 0x000001686A1C95D0>, {'enabled': False, 'model_type': 'iclight_sd15_fc_unet_ldm', 'input_fg': None, 'uploaded_bg': None, 'bg_source_fc': 'None', 'bg_source_fbc': 'Use Background Image', 'remove_bg': True, 'reinforce_fg': False, 'detail_transfer': False, 'detail_transfer_use_raw_input': False, 'detail_transfer_blur_radius': 5}, 0, ControlNetUnit(input_mode=<InputMode.SIMPLE: 'simple'>, use_preview_as_input=False, batch_image_dir='', batch_mask_dir='', batch_input_gallery=[], batch_mask_gallery=[], multi_inputs_gallery=[], generated_image=None, mask_image=None, hr_option=<HiResFixOption.BOTH: 'Both'>, enabled=False, module='None', model='None', weight=1, image=None, resize_mode=<ResizeMode.INNER_FIT: 'Crop and Resize'>, processor_res=-1, threshold_a=-1, threshold_b=-1, guidance_start=0, guidance_end=1, pixel_perfect=False, control_mode=<ControlMode.BALANCED: 'Balanced'>, advanced_weighting=None, save_detected_map=True), ControlNetUnit(input_mode=<InputMode.SIMPLE: 'simple'>, use_preview_as_input=False, batch_image_dir='', batch_mask_dir='', batch_input_gallery=[], batch_mask_gallery=[], multi_inputs_gallery=[], generated_image=None, mask_image=None, hr_option=<HiResFixOption.BOTH: 'Both'>, enabled=False, module='None', model='None', weight=1, image=None, resize_mode=<ResizeMode.INNER_FIT: 'Crop and Resize'>, processor_res=-1, threshold_a=-1, threshold_b=-1, guidance_start=0, guidance_end=1, pixel_perfect=False, control_mode=<ControlMode.BALANCED: 'Balanced'>, advanced_weighting=None, save_detected_map=True), ControlNetUnit(input_mode=<InputMode.SIMPLE: 'simple'>, use_preview_as_input=False, batch_image_dir='', batch_mask_dir='', batch_input_gallery=[], batch_mask_gallery=[], multi_inputs_gallery=[], generated_image=None, mask_image=None, hr_option=<HiResFixOption.BOTH: 'Both'>, enabled=False, module='None', model='None', weight=1, image=None, resize_mode=<ResizeMode.INNER_FIT: 'Crop and Resize'>, processor_res=-1, threshold_a=-1, threshold_b=-1, guidance_start=0, guidance_end=1, pixel_perfect=False, control_mode=<ControlMode.BALANCED: 'Balanced'>, advanced_weighting=None, save_detected_map=True), False, 7, 1, 'Constant', 0, 'Constant', 0, 1, 'enable', 'MEAN', 'AD', 1, False, 1.01, 1.02, 0.99, 0.95, False, 0.5, 2, False, 256, 2, 0, False, False, 3, 2, 0, 0.35, True, 'bicubic', 'bicubic', False, 0, 'anisotropic', 0, 'reinhard', 100, 0, 'subtract', 0, 0, 'gaussian', 'add', 0, 100, 127, 0, 'hard_clamp', 5, 0, 'None', 'None', False, 'MultiDiffusion', 768, 768, 64, 4, False, False, False, False, False, 'positive', 'comma', 0, False, False, 'start', '', 1, '', [], 0, '', [], 0, '', [], True, False, False, False, False, False, False, 0, False) {}
    Traceback (most recent call last):
      File "S:\stable-diffusion-webui-reForge\modules\call_queue.py", line 74, in f
        res = list(func(*args, **kwargs))
    TypeError: 'NoneType' object is not iterable

**Reply by Panchovix** [reply]
Oh I see, give me some time since I'm a bit busy right now on work, but will have a look on it.

**Reply by Panchovix** [reply, marked as answer]
Updated them now and they seem to work fine now. Can you test and see how it goes? If it fails specifically using controlnet it may be from another reason.

**Reply by pinea00** [reply]
thx~it work normally
